### PR TITLE
Introduce fce test

### PR DESCRIPTION
**Pull request by michaelvoronov**
Linked PR: https://github.com/fluencelabs/fce/pull/69

### CODE REVIEW

File: crates/fce-test-macro-impl/src/fce_test/glue_code_generator.rs [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+///
+/// let tmp_dir = tmp_dir.join(&service_id);
+/// let tmp_dir = tmp_dir.to_string_lossy().to_string();
+/// std::fs::create_dir(&tmp_dir).expect("can't create a directory for service in tmp");
+///
+/// let mut __fce__generated_fce_config = fluence_test::internal::TomlAppServiceConfig::load("/path/to/greeting/Config.toml".to_string())
+///     .unwrap_or_else(|e| {
+///         panic!(
+///              "app service located at `{}` config can't be loaded: {}",
+///            "/Users/mike/dev/work/fluence/wasm/fce/examples/greeting/Config.toml", e
```
Comment (folex) [inline crates/fce-test-macro-impl/src/fce_test/glue_code_generator.rs:71]:
```suggestion
///            "/path/to/greeting/Config.toml", e
```

File: crates/fce-test-macro-impl/src/fce_test/glue_code_generator.rs [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+///
+/// let mut greeting = __fce_generated_greeting::FCEGeneratedStructgreeting::new(fce);
+///
+/// // (3)
+///```
+///
+/// Here [(0), (1)] - module_definitions
+///      [(1), (2)] - AppService ctor
+///      [(2), (3)] - ctors of all modules of the tested service
+///      [(3), (4)] - original test function
```
Comment (folex) [inline crates/fce-test-macro-impl/src/fce_test/glue_code_generator.rs:95]: Where's `(4)`?

Comment (folex) [inline crates/fce-test-macro-impl/src/fce_test/glue_code_generator.rs:95]: I was confused by this block. Maybe modify it like that? (+ see below modifications in code to make it clearer)

```suggestion
/// Example code above corresponds to the macro definition in the following way:
///      [(0), (1)] - module_definitions*
///      [(1), (2)] - app_service_ctor
///      [(2), (3)] - module_ctors*
///      [(3), (4)] - original_block
```

File: crates/fce-test-macro-impl/src/fce_test/glue_code_generator.rs [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    let glue_code = quote! {
+        #[test]
+        #signature {
+            #(#module_definitions)*
+
+            #app_service_ctor
+
+            #(#module_ctors)*
+
+            #original_block
```
Comment (folex) [inline crates/fce-test-macro-impl/src/fce_test/glue_code_generator.rs:127]:
```suggestion
            // definitions for wasm modules specified in config
            #(#module_definitions)*
            // AppService constructor and instantiation to implicit `fce` variable
            #app_service_ctor
            // constructors of all modules of the tested service
            #(#module_ctors)*
            // original test function as is
            #original_block
```

File: crates/fce-test-macro-impl/src/fce_test/module_generator/methods_generator.rs [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+            Ok(output_type)
+        }
+    }
+}
+
+fn get_output_type(output_types: &[IType]) -> Option<&IType> {
+    match output_types.len() {
+        0 => None,
+        1 => Some(&output_types[0]),
+        _ => unimplemented!("function with more than 1 arguments aren't supported now"),
```
Comment (folex) [inline crates/fce-test-macro-impl/src/fce_test/module_generator/methods_generator.rs:159]: Should it be `more than 1 outputs/return values`?

**Review by folex — APPROVED** [review]

File: crates/fce-test-macro-impl/src/fce_test/module_generator/methods_generator.rs (hunk 4 above)
Comment (michaelvoronov) [reply crates/fce-test-macro-impl/src/fce_test/module_generator/methods_generator.rs:159]: Only after switching to a VM with multi-value

**Pull request merged**